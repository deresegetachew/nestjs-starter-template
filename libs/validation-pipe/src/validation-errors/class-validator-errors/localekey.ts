export enum localeKey {
    isNotEmpty = 'dto:is-not-empty',
    minLength = 'dto:min-length',
    isEmail = 'dto:is-email',
    isString = 'dto:is-string'
}