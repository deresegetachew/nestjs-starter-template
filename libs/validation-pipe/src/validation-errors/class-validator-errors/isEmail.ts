import { I18nError, LogLevel } from '@lib/common';
import { BadRequestException } from "@nestjs/common";
import { localeKey } from './localekey';


class IsEmailI18n extends I18nError {
    constructor(message: string, variables: { [key: string]: string | number } = {}) {
        super(message, variables, LogLevel.Info);
    }
}

export class IsEmail extends BadRequestException {
    public I18nError: IsEmailI18n;
    constructor(email: string) {
        super();
        this.I18nError = new IsEmailI18n(localeKey.isEmail, { email });
    }
}


