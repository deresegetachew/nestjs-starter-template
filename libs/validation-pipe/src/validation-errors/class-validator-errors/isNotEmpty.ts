import { I18nError, LogLevel } from '@lib/common';
import { BadRequestException } from "@nestjs/common";
import { localeKey } from './localekey';

class IsNotEmptyI18n extends I18nError {
    constructor(message: string, variables: { [key: string]: string | number } = {}) {
        super(message, variables, LogLevel.Info);
    }
}
export class IsNotEmpty extends BadRequestException {
    public I18nError: IsNotEmptyI18n;

    constructor(field: string) {
        super();
        this.I18nError = new IsNotEmptyI18n(localeKey.isNotEmpty, { field });
    }
}


