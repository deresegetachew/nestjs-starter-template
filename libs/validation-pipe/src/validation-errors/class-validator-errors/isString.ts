import { I18nError, LogLevel } from '@lib/common';
import { BadRequestException } from "@nestjs/common";
import { localeKey } from './localekey';

class IsStringI18n extends I18nError {
    constructor(message: string, variables: { [key: string]: string | number } = {}) {
        super(message, variables, LogLevel.Info);
    }
}

export class IsString extends BadRequestException {
    public I18nError: IsStringI18n;

    constructor(field: string) {
        super();
        this.I18nError = new IsStringI18n(localeKey.isString, { field });
    }
}

