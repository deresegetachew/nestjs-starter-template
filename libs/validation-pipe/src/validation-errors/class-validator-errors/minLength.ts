import { I18nError, LogLevel } from '@lib/common';
import { BadRequestException } from "@nestjs/common";
import { localeKey } from './localekey';

class MinLengthI18n extends I18nError {
    constructor(message: string, variables: { [key: string]: string | number } = {}) {
        super(message, variables, LogLevel.Info);
    }
}
export class MinLength extends BadRequestException {
    public I18nError: MinLengthI18n;
    constructor(field: string, length: number) {
        super();
        this.I18nError = new MinLengthI18n(localeKey.minLength, { field, length });
    }
}


