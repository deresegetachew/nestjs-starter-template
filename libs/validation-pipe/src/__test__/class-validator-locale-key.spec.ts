import { localeKey } from '../validation-errors/class-validator-errors/localekey';

describe('classValidator error Locale keys', () => {
    it('matches enum definition snapshot', () => {
        const keys = JSON.stringify(localeKey);
        expect(keys).toMatchSnapshot();
    })
})