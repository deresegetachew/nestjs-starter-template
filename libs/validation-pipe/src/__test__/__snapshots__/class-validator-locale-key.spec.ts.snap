// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classValidator error Locale keys matches enum definition snapshot 1`] = `"{\\"isNotEmpty\\":\\"dto:is-not-empty\\",\\"minLength\\":\\"dto:min-length\\",\\"isEmail\\":\\"dto:is-email\\",\\"isString\\":\\"dto:is-string\\"}"`;
